### PR TITLE
Update to Bevy 0.11

### PR DESCRIPTION
**Pull request by DaylightNebula**
I have found this library very useful with Bevy 0.10, however, Bevy 0.11 was just released.  So here is my solution to update to Bevy 0.11 following the [migration guide](https://bevyengine.org/learn/migration-guides/0.10-0.11/).

### CODE REVIEW

**Comment by sparten11740** [comment]
thanks for your work - I was also meaning to get to this but discovered a bug in the multi-bar scenario where only the bar would get updated that had its plugin registered first. I first thought this had something to do with the bevy upgrade but it's also present in 0.10. I will merge #15 so this lib is available for `0.11.0` and maybe we can continue with the web fixes here

File: src/material.rs [hunk 1]
```diff
@@ -83,3 +79,13 @@ impl Material for BarMaterial {
...
+
+impl TypePath for BarMaterial {
+    fn type_path() -> &'static str {
+        "bevy_health_bar3d::BarMaterial"
+    }
+
+    fn short_type_path() -> &'static str {
+        "bevy_health_bar3d::BarMaterial"
+    }
+}
```
Comment (sparten11740) [inline src/material.rs:91]: this can be derived either indirectly by deriving `Reflect` or directly by deriving `TypePath`. The short type path of the derived version would return `BarMaterial` and the full path `bevy_health_bar3d::material::BarMaterial`

File: src/plugin.rs [hunk 2]
```diff
@@ -22,18 +23,18 @@ impl<T: Percentage + Component> Default for HealthBarPlugin<T> {
     }
 }
 
-impl<T: Percentage + Component> Plugin for HealthBarPlugin<T> {
+impl<T: Percentage + Component + TypePath> Plugin for HealthBarPlugin<T> {
```
Comment (sparten11740) [inline src/plugin.rs:26]: I would like to avoid adding this constraint on `T` (which I did in my PR). Disadvantage is that the registered types have to go for now. I am hoping for a solution on the bevy side that allows falling back to `std::any::type_name`

File: src/plugin.rs [hunk 3]
```diff
@@ -106,15 +109,18 @@ fn spawn<T: Percentage + Component>(
             let border = border.unwrap_or(&default_border);
 
             let material = materials.add(BarMaterial {
-                value: percentage.value(),
+                // settings: Vec4 { x: percentage.value(), y: border_width, z: width, w: height },
+                // settings: Vec4(percentage.value(), border_width, width, height),
+                settings: (percentage.value(), border.width, width, height).into(),
```
Comment (sparten11740) [inline src/plugin.rs:114]: all of these are settings of sorts. maybe a little arbitrary to pack these 4 in a vector. why is that needed?

Comment (DaylightNebula) [reply src/plugin.rs:114]: When I tried to run a game using this on the web.  I was getting the error mentioned [here](https://github.com/gfx-rs/wgpu/issues/2832).  My fix was to make everything 16 byte aligned by converting all of the shaders inputs to vec4's.

File: src/material.rs (hunk 1 above)
Comment (DaylightNebula) [reply src/material.rs:91]: Yes that is my bad.  I forgot I could do that when I saw that TypePath was now required in 0.11.

File: src/plugin.rs (hunk 2 above)
Comment (DaylightNebula) [reply src/plugin.rs:26]: I did that to fix a compilation error that occurs in the register_type calls before that say type path is required to register BarWidth or BarOffset.

File: src/plugin.rs (hunk 3 above)
Comment (sparten11740) [reply src/plugin.rs:114]: IC, makes sense. web wasn't on my radar, thanks for bringing it to my attention. I ported part of your fix to #17

**Comment by sparten11740** [comment]
mind closing here as this is supported upstream?
